github/client_repository_file.go

```diff
@@ -61,17 +74,26 @@ func (c *FileClient) Get(ctx context.Context, path, branch string, optFns ...git
 	for _, file := range directoryContent {
 		filePath := file.Path
 		if *file.Type == "dir" {
-			if recursive == true {
+			if fileOpts.Recursive == true {
+				// stop recursive calls when level is the max level reached
+				if fileOpts.MaxDepth == 1 {
```

Comment:
- `MaxDepth == 0` is treated the same way as `MaxDepth == 1` where the files in the path are returned no nested levels. I made that assumption because if 0 was provided why do we need to call the Get in the first place. If that is to be changed the default value of the maxDepth should be changed according to what?

- Yes, the break should be replaced with a continue, where only the recursive call is skipped but the rest of the files are retrieved in the current path

Reply:
it should be `fileOpts.MaxDepth <= 1` then?

Even if you replace `break` by `continue` you won't reach `output, _, err := c.c.Client().Repositories.DownloadContents(ctx, c.ref.GetIdentity(), c.ref.GetRepository(), *filePath, opts)` for the filePath on the last level

Reply:
but each iteration is either a directory or file, and if it is a directory there is nothing to download